### PR TITLE
macOS contacts: prefer personal over app generated

### PR DESCRIPTION
**Pull request by georgehrke**
https://sabre.io/dav/clients/osx-addressbook/

This PR is adding support for ctag:
> This client requires impementation of the proprietary '{http://calendarserver.org/ns/}getctag' property. Without it, it will simply not request any vcards.

This PR renames `app-generated` to `z-app-generated`:
> OS X addressbook can only ever work with 1 address book. Even though the CardDAV standard makes it easy for users to own more than one addressbook, this is not supported.
> 
> Unfortunately this means that if a user has more than one addressbook in their account, only the 'first' will show up, and contacts from other addressbooks are completely hidden to the user.
> 
> This can cause for a lot of confusion, and there is no obvious solution other than simply enforce a 1 address book limit for users.

tl;dr:
Without this PR, macOS Contacts will only show the 5 contacts from the contacts interaction addressbook, but all the contacts from the personal addressbook will be gone.

### CODE REVIEW

**Comment by georgehrke** [comment]
While the sabre/dav docs say that the one addressbook limitation affect only `OS X 10.6 until OS X 10.10`, i'm still affected by it on macOS 10.15. So I guess the docs are not up to date.

**Review by rullzer — APPROVED** [review]
Fine by me

**Review by ChristophWurst — APPROVED** [review]

**Comment by georgehrke** [comment]
Test failures unrelated, merging
